osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/DlmsConnectionMessageProcessor.java

```diff
@@ -52,11 +53,19 @@
     @Autowired
     protected DlmsDeviceRepository deviceRepository;
 
-    protected DlmsConnectionManager createConnectionForDevice(final DlmsDevice device,
+    @Autowired
+    protected ThrottlingService throttlingService;
+
+    public DlmsConnectionManager createConnectionForDevice(final DlmsDevice device,
             final MessageMetadata messageMetadata) throws OsgpException {
+
+        throttlingService.newConnectionRequest();
+        throttlingService.openConnection();
```

Comment:
Ah ja, hier de toepassing. Beide call zijn dus inderdaad nodig. Wanneer openConnection() de verantwoordelijkheid neem om newConnectionRequest aan te roepen hou je hier maar 1 call over. De ThrottlingService heeft dan de verantwoordelijkheid een connection op te leveren, de implementatie van hoe die dat doet is encapsulated.

Reply:
ja daar heb ik idd ook aan zitten denken, op deze manier is het echter explicieter wat er gebeurd (eerst request en dan openConnection) door de twee poortjes achter elkaar te zetten in de client

Reply:
done